ext/standard/php_var.h

```diff
@@ -59,6 +59,8 @@ PHPAPI zend_long php_var_unserialize_get_cur_depth(php_unserialize_data_t d);
 #define PHP_VAR_UNSERIALIZE_DESTROY(d) \
 	php_var_unserialize_destroy(d)
 
+PHPAPI void unserialize_with_options(zval *return_value, const char *buf, const size_t buf_len, zval *options, const char* function_name);
```

Comment:
Creating a new API seemed to be the best way to ensure that `Phar*->getMetadata()` would stay in sync with the unserialize() implementation.

Do you have alternative suggestions for names or where to locate it, additional comments on the signature, etc?

Reply:
New API is good, but please add a `php_` prefix.

Reply:
It would also be better to accept `HashTable *options` instead. You can use Z_PARAM_ARRAY_HT to fetch it.